Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -642,6 +642,20 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
 
 }
 
+extension UUID {
+    /// Copies `self` into a `Binary`.
+    internal func asBinary() throws -> Binary {
```

Comment:
That does seem cleaner, as otherwise this is introducing a generically named `asBinary` method on the core UUID class, which returns an instance of _our_ Binary object.